### PR TITLE
complete feature login for rent car website

### PR DESCRIPTION
**Pull request by khajackie2206**

### CODE REVIEW

File: src/core/Response.php [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+    /**
+     * @param array $data
+     * @param int $statusCode
+     * @return $this
+     */
+    public function toJson(array $data, int $statusCode = self::HTTP_OK): self
+    {
+        $this->setStatusCode($statusCode);
+        $this->setData($data);
+        /*$this->headers = array_merge($this->headers, [
```
Comment (tolehoai) [inline src/core/Response.php:28]: remove your comment code

File: src/validation/CarValidator.php [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    /**
+     * @param CarTransfer $car
+     * @return array|Validator
+     */
+    public function validateCar(CarTransfer $car)
+    {
+        $validator = new Validator();
+        $validator->name('brand')->value($car->getBrand())->min(3)->max(255)->required();
+        $validator->name('color')->value($car->getColor())->max(30)->required();
+        $validator->name('price')->value($car->getPrice())->is_int()->max(5000)->required();
```
Comment (tolehoai) [inline src/validation/CarValidator.php:19]: max 5000, it's too small for price

File: src/validation/ImageValidator.php [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+            "jpeg" => "image/jpeg",
+            "gif" => "image/gif",
+            "png" => "image/png"
+        );
+        $ext = pathinfo($image["name"], PATHINFO_EXTENSION);
+        if (!array_key_exists($ext, $allowed) || !in_array($image['type'], $allowed)) {
+            return ['error' => "Please select a valid file format!"];
+        }
+        $maxsize = 10 * 1024 * 1024;
+        if ($image["size"] > $maxsize) {
```
Comment (tolehoai) [inline src/validation/ImageValidator.php:27]: use single quote

File: src/validation/UserValidator.php [hunk 4]
```diff
@@ -0,0 +1,35 @@
+<?php
```
Comment (tolehoai) [inline src/validation/UserValidator.php:1]: should split into LoginValidator and RegisterValidator

File: src/validation/Validator.php [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+    {
+        if ((isset($this->file)) || ($this->value == '' || $this->value == null)) {
+            $this->errors[$this->name] = 'Field value ' . $this->name . ' is required.';
+        }
+        return $this;
+    }
+
+    public function min($length)
+    {
+        if (is_string($this->value)) {
```
Comment (tolehoai) [inline src/validation/Validator.php:80]: if else if else if ?? too much

File: public/index.php [hunk 6]
```diff
@@ -1,11 +1,13 @@
...
+
 session_start();
-use Khanguyennfq\CarForRent\core\Application;
-require_once __DIR__.'/../vendor/autoload.php';
+
+use Khanguyennfq\CarForRent\app\Application;
+use Khanguyennfq\CarForRent\core\Request;
+require_once __DIR__ . '/../vendor/autoload.php';
 
 error_reporting(E_ALL);
```
[the remark this answers is not in the document]
Comment (khajackie2206) [reply public/index.php:9]: oke thank you, I removed it

File: src/validation/Validator.php (hunk 5 above)
Comment (khajackie2206) [reply src/validation/Validator.php:80]: oke I will fix it

File: src/validation/CarValidator.php (hunk 2 above)
Comment (khajackie2206) [reply src/validation/CarValidator.php:19]: It just rent cost, not for buy a new car to set higher cost

File: public/index.php [hunk 7]
```diff
@@ -1,11 +1,13 @@
...
 ini_set('display_errors', 1);
```
[the remark this answers is not in the document]
Comment (khajackie2206) [reply public/index.php:10]: oke bro

**Pull request closed without merging**